vault/vault.go

```diff
@@ -612,8 +612,16 @@ func (v *Vault) Copy(oldpath, newpath string, opts MoveCopyOpts) error {
 		}
 	}
 
-	srcPath, srcKey, _ := ParsePath(oldpath)
-	dstPath, dstKey, _ := ParsePath(newpath)
+	srcPath, srcKey, srcVersion := ParsePath(oldpath)
+	dstPath, dstKey, dstVersion := ParsePath(newpath)
+
+	if dstVersion != 0 {
+		return fmt.Errorf("Copying a secret to a specific destination version is not supported")
+	}
+
+	if opts.Deep && srcVersion != 0 {
+		return fmt.Errorf("Performing a deep copy of a specified version is not supported")
+	}
 
```

Comment:
extra constraint, recursion and a specific version doesn't make much sense either